pkg/cli/zip_test.go

```diff
@@ -1252,3 +1252,60 @@ func TestCommandFlags(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// This tests the operation of zip over excluded nodes.
+func TestPartialZipForExcludedNodes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	dir, cleanupFn := testutils.TempDir(t)
+	defer cleanupFn()
+	c := NewCLITest(TestCLIParams{
+		StoreSpecs: []base.StoreSpec{{
+			Path: dir,
+		}},
+	})
+	defer c.Cleanup()
+	zipName := filepath.Join(dir, "debug.zip")
+
+	// We want a low timeout so that the test doesn't take forever;
+	// however low timeouts make race runs flaky with false positives.
+	skip.UnderShort(t)
+	skip.UnderRace(t)
+
+	sc := log.ScopeWithoutShowLogs(t)
+	defer sc.Close(t)
+	// Reduce the number of output log files to just what's expected.
+	defer sc.SetupSingleFileLogging()()
+
+	ctx := context.Background()
+
+	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
+			Insecure:          true,
+		},
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	_, err := c.RunWithCapture("debug zip --concurrency=1 --exclude-nodes=1 --cpu-profile-duration=0 " + dir + "/debug.zip")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r, err := zip.OpenReader(zipName)
+	defer func() { _ = r.Close() }()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	d, err := r.Open("debug/nodes/1")
+	defer func() {
+		if d != nil {
+			_ = d.Close()
+		}
+	}()
+	if err == nil {
+		t.Fatal("node directory should not be present in the zip", err)
+	}
```

Comment:
We should probable check for that exact error. We should fail the test if there is any other error.

```suggestion
  require.True(t, os.IsNotExist(err), "node directory should not be present in the zip")
```

Reply:
DONE
<!-- Reviewable comment -OB9hoHB2VJWTI9EZUEB:bb6tyiv -->
<!-- Sent from Reviewable.io -->